lasagne/layers/normalization.py

```diff
@@ -373,3 +377,225 @@ def batch_norm(layer, **kwargs):
         nonlin_name = bn_name and bn_name + '_nonlin'
         layer = NonlinearityLayer(layer, nonlinearity, name=nonlin_name)
     return layer
+
+
+class StandardizationLayer(Layer):
+    """
+    This layer implements the normalization of the input layer's
+    outputs across the specified axes:
+
+    .. math::
+        y_i = \\frac{x_i - \\mu_i}{\\sqrt{\\sigma_i^2 + \\epsilon}}
+
+    That is, each input feature (or input pixel) :math:`\\x_i`
+    is normalized to zero mean and unit variance.
+    The mean :math:`\\mu_i` and variance
+    :math:`\\sigma_i^2` is calculated across the specified axes.
+    In contrast to batch normalization, the mean and
+    variance is not restricted to be defined across examples,
+    so the same operation can be applied during training and testing.
+    The advantages of using this implementation over e.g.
+    :class:`BatchNormLayer` with adapted axes arguments, are its
+    independence of the input size, as no parameters are learned and stored.
+    :class:`StandardizationLayer` can be employed to realize
+    different normalization schemes such as instance normalization [1]
+    and layer normalization [2], for both of which convenience functions
+    (:func:`instance_norm` and :func:`layer_norm`) are available.
+
+    Parameters
+    ----------
+    incoming : a :class:`Layer` instance or a tuple
+        The layer feeding into this layer, or the expected input shape
+    axes : 'auto', int or tuple of int
+        The axis or axes to normalize over. If ``'auto'`` (the default),
+        normalize over all axes except for the first two:
+        this will normalize over all spatial dimensions for
+        convolutional layers.
+    epsilon : scalar
+        Small constant :math:`\\epsilon` added to the variance before taking
+        the square root and dividing by it, to avoid numerical problems
+    **kwargs
+        Any additional keyword arguments are passed to the :class:`Layer`
+        superclass.
+        
+    Notes
+    -----
+    This layer should be inserted between a linear transformation (such as a
+    :class:`DenseLayer`, or :class:`Conv2DLayer`) and its nonlinearity. The
+    convenience functions :func:`instance_norm` :func:`layer_norm`
+    modify an existing layer to insert instance normalization or
+    layer normalization in front of its nonlinearity.
+
+    See also
+    --------
+    instance_norm : Convenience function to apply instance normalization to a layer
+    References
+    layer_norm : Convenience function to apply layer normalization to a layer
+    References
+
+    ----------
+    .. [1] Ulyanov, D., Vedaldi, A., & Lempitsky, V. (2016):
+           Instance Normalization: The Missing Ingredient for Fast Stylization.
+           https://arxiv.org/pdf/1607.08022.pdf.
+
+    .. [2] Ba, J., Kiros, J., & Hinton, G. (2016):
+           Layer normalization. arXiv preprint arXiv:1607.06450.
```

Comment:
Let's unify this and give the arxiv link in both cases, but using `abs` instead of `pdf`.